deploy/runtools/run_farm_deploy_managers.py

```diff
@@ -172,12 +239,21 @@ def copy_sim_slot_infrastructure(self, slotno: int) -> None:
             run(f"mkdir -p {remote_sim_rsync_dir}")
 
             files_to_copy = serv.get_required_files_local_paths()
-            for local_path, remote_path in files_to_copy:
-                # -z --inplace
-                rsync_cap = rsync_project(local_dir=local_path, remote_dir=pjoin(remote_sim_rsync_dir, remote_path),
-                            ssh_opts="-o StrictHostKeyChecking=no", extra_opts="-L", capture=True)
-                rootLogger.debug(rsync_cap)
-                rootLogger.debug(rsync_cap.stderr)
+
+            with TemporaryDirectory() as uridir:
+                # Download all URIs needed. When no download is required, 
+                # an empty list is returned.
+                # All of this code is inside a with block so that the 
+                # downloads will survive until after the rsync below,
+                # at which point the local copies are not needed and removed.
+                files_to_copy.extend(self.local_download_all_uri(slotno, uridir))
+
+                for local_path, remote_path in files_to_copy:
+                    # -z --inplace
+                    rsync_cap = rsync_project(local_dir=local_path, remote_dir=pjoin(remote_sim_rsync_dir, remote_path),
+                                ssh_opts="-o StrictHostKeyChecking=no", extra_opts="-L", capture=True)
+                    rootLogger.debug(rsync_cap)
+                    rootLogger.debug(rsync_cap.stderr)
```

Comment:
I was hoping that when we added support for URIs that we would not download everything to the manager than then copy it to the runtime hosts, we would do the downloading in a distributed fashion on the runtime hosts.  When running large simulations of dozens of nodes, we're already constrained by the bandwidth available on the single manager host.

It also seems like this will change repeated `infrasetup` runs from being a checksum check done by the rsync to re-downloading the local copy and then doing the checksum to see that it doesn't need to be copied remotely.

If downloading locally makes it easier to add the URI support, we can distribute the downloading later to address the bottleneck.

Reply:
Benjamin pointed out my flawed understanding of `fabric.api.execute()`.  Distributing the downloads is out of scope for this PR.